Home_5/School.cs

```diff
@@ -0,0 +1,141 @@
+        /// Создайте метод вывода в консоль студента из каждой группы с наилучшей оценкой по Биологии. Если существуют студенты с одинаковыми наилучшими оценками - выведите любого из них.
+        /// Установите каждому из лучших студентов выше - денежное вознаграждение в рублях - reward (rand: 0 - 100) (предусмотрите, чтобы нельзя было установить значение reward< 1 рубля)
+        /// Создайте методы подсчета и вывода среднего балла группы студентов по каждой из дисциплин и вывода этой информации в консоль (разные методы) (ex: Group1, avarage mark math: 8.3)
+        /// Создайте метод расчета среднего балла группы студентов по всем 3 дисциплинам(средняя оценка группы по каждой дисциплине / количество дисциплин) и выведите эту информацию в консоль.
+        /// (Avearage mark of Group1 - Math, PhysicalEducation, Biology - 8.3)
+        /// Добавьте каждому студенту из группы с наибольшим средним баллом по всем дисциплинам произвольный reward.
+        /// Выведите на экран студента с наибольшим reward. Если таких студентов несколько - выведите их всех.
+        /// </summary>
+
+        private static int amountOfGroupsInSchool = 3;
+        
+        public StudentsGroup[] SchoolGroups { get; set; }
+        
+        public School()
+        {
+            StudentsGroup[] groupsAllTogether = new StudentsGroup[amountOfGroupsInSchool];
+
+            for (int i = 0; i < groupsAllTogether.Length; i++)
+            {
+                groupsAllTogether[i] = new StudentsGroup();
+            }
+
+            SchoolGroups = groupsAllTogether;
+        }
+        
+        public void bestMarksInEachGroup()
+        {
+            foreach (StudentsGroup studentsGroup in this.SchoolGroups)
+            {
+                Student bestMathMarkStudent = studentsGroup.BestMathMarkInGroup();
+                Student bestPhysicalEducationMark = studentsGroup.BestPhysicalEducationMarkInGroup();
+                Student bestBiologyMarkStudent = studentsGroup.BestBiologyMarkInGroup();
+
+                Console.WriteLine($"Group{studentsGroup.GroupNumber}:");
+                Console.WriteLine($"Best Math mark: {bestMathMarkStudent.Name}, Mark: {bestMathMarkStudent.MathMark}");
+                Console.WriteLine($"Best Physical Education mark: {bestPhysicalEducationMark.Name}, Mark: {bestPhysicalEducationMark.PhysicalEducationMark}");
+                Console.WriteLine($"Best Biology mark: {bestBiologyMarkStudent.Name}, Mark: {bestBiologyMarkStudent.BiologyMark}\n");
+
+                RewardProvision(bestMathMarkStudent);
+                RewardProvision(bestPhysicalEducationMark);
+                RewardProvision(bestBiologyMarkStudent);
+                Console.WriteLine();
+            }
+        }
+
+        public void averageMarksInEachGroup()
+        {
+            foreach (StudentsGroup studentsGroup in this.SchoolGroups)
```

Comment:
You can use just SchoolGroups, instead of this SchoolGroups

Reply:
fixed